File: backend/src/main/java/com/example/starter_project_2025/system/course/entity/Material.java
package com.example.starter_project_2025.system.course.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import com.example.starter_project_2025.system.course.enums.MaterialType;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "materials")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Material {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(length = 500)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MaterialType type;

    @Column(nullable = false)
    private String sourceUrl;

    @Column(length = 1000)
    private String tags;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id", nullable = false)
    private Session session;

    @Column(nullable = false)
    private Integer displayOrder = 0;

    @Column(nullable = false)
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: frontend/src/types/material.ts
export type MaterialType = 'VIDEO' | 'DOCUMENT' | 'LINK' | 'IMAGE' | 'AUDIO';

export interface Material {
  id: string;
  title: string;
  description?: string;
  type: MaterialType;
  sourceUrl: string;
  tags?: string;
  sessionId: string;
  displayOrder: number;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateMaterialRequest {
  title: string;
  description?: string;
  type: MaterialType;
  sourceUrl: string;
  tags?: string;
  sessionId: string;
  displayOrder?: number;
  isActive?: boolean;
}

export interface UpdateMaterialRequest {
  title?: string;
  description?: string;
  type?: MaterialType;
  sourceUrl?: string;
  tags?: string;
  displayOrder?: number;
  isActive?: boolean;
}

export const MATERIAL_TYPE_OPTIONS = [
  { value: 'VIDEO', label: 'Video' },
  { value: 'DOCUMENT', label: 'Document' },
  { value: 'LINK', label: 'Link' },
  { value: 'IMAGE', label: 'Image' },
  { value: 'AUDIO', label: 'Audio' },
] as const;

export type MaterialTypeValue = (typeof MATERIAL_TYPE_OPTIONS)[number]['value'];
